r/src/compute-exec.cpp

```diff
@@ -302,6 +301,8 @@ std::shared_ptr<acero::ExecNode> ExecNode_Scan(
       "make_struct", std::move(exprs),
       compute::MakeStructOptions{cpp11::as_cpp<std::vector<std::string>>(field_names)});
 
+  options->dataset_schema = GetProjectedSchemaFromExpression(options->projection, dataset->schema());
```

Comment:
I don't think we can actually do this this way as this function is internal, but I think that we need to do something like what this does.